execution/v0.1.1/interface.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 Cloudbase Solutions SRL
+//
+//    Licensed under the Apache License, Version 2.0 (the "License"); you may
+//    not use this file except in compliance with the License. You may obtain
+//    a copy of the License at
+//
+//         http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+//    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+//    License for the specific language governing permissions and limitations
+//    under the License.
+
+package executionv011
+
+import (
+	"context"
+
+	"github.com/cloudbase/garm-provider-common/params"
+)
+
+// ExternalProvider defines an interface that external providers need to implement.
+// This is very similar to the common.Provider interface, and was redefined here to
+// decouple it, in case it may diverge from native providers.
+type ExternalProvider interface {
+	// CreateInstance creates a new compute instance in the provider.
+	CreateInstance(ctx context.Context, bootstrapParams params.BootstrapInstance) (params.ProviderInstance, error)
```

Comment:
It seems that the signatures for the common functions both this version and the previous version, are identical. It makes sense to create a common interface with the common functions, and embed that common interface in both this one and the previous version.

If the previous version is identical to the base one, you can just create an alias to the base interface. Or just use the base interface directly.